File: Problem7/code.js
const isPrime = (num) => {
  let response = true;
  if (num <= 1) {
    response = false;
  }
  for (let i = 2, s = Math.floor(Math.sqrt(num)); i <= s; i++)
    if (num % i === 0) {
      response = false;
    }
  return response;
};

let giveMePrimeN = (num) => {
  let count = 0;
  let n = 2;
  let arrOfPrimes = [];
  while (count < num) {
    if (isPrime(n)) {
      count++;
      arrOfPrimes.push(n);
    }
    n++;
  }

  return [arrOfPrimes, arrOfPrimes[arrOfPrimes.length-1]];
};

console.log(giveMePrimeN(10001));

File: Problem7/code.py
import math

def isPrime(num):
    if num % 2 == 0:
        return False
    if num > 1:
        
        num_sqrt = int(math.ceil(math.sqrt(num)))
        
        for i in range(2, num_sqrt):
            if (num % i) == 0:
                response = False
                break
        else:
            response = True
        return response
            

def give_me_prime_n(nth):

    prime_list = []
    n = 0
    
    while len(prime_list) <= nth:
        if isPrime(n):
            prime_list.append(n)
        n += 1  
            
    print(prime_list[-1])

give_me_prime_n(10001) #104743
